feature-policy/resources/picture-in-picture.js

```diff
@@ -3,14 +3,18 @@ function isPictureInPictureAllowed() {
     let video = document.createElement('video');
     video.src = '/media/movie_5.ogv';
     video.onloadedmetadata = () => {
-      video.requestPictureInPicture()
-      .then(() => resolve(document.pictureInPictureEnabled))
-      .catch(e => {
-        if (e.name == 'NotAllowedError')
-          resolve(document.pictureInPictureEnabled);
-        else
-          resolve(false);
-      });
+      try {
+        video.requestPictureInPicture()
+        .then(() => resolve(document.pictureInPictureEnabled))
+        .catch(e => {
+          if (e.name == 'NotAllowedError')
+            resolve(document.pictureInPictureEnabled);
+          else
+            resolve(false);
+        });
+      } catch(e) {
```

Comment:
I don't know much about feature policy or picture in picture, but this broad exception catch smells a bit. Are you trying to catch exceptions from `video.requestPictureInPicture` (e.g. undefined)?

Reply:
Yeah. I suppose I can do `t.step_func` instead.

Reply:
The function is sometimes called when there's no Test object. But I've changed to an `if` check instead.